sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubsModelFactory.cs

```diff
@@ -291,7 +291,12 @@ public override bool TryAdd(EventData eventData)
             ///
             /// <returns>The set of events as an enumerable of the requested type.</returns>
             ///
-            public override IEnumerable<T> AsEnumerable<T>() => (IEnumerable<T>)_backingStore;
+            public override List<T> AsList<T>() => _backingStore switch
```

Comment:
The model factory method takes `IList<EventData>`.   Since this class is intended for testing purposes, I didn't think adding a new factory method and hiding the existing one was warranted.  Instead, we're just paying for a list allocation.